vagrant/README.md

```diff
@@ -18,16 +18,35 @@ Currently it uses Kubernetes v1.6.1
-cache and Docker images. This allows you to reuse those caches on subsequent
-provisioning of the VMs, and it is intended to help situations where one is
-developing in an environment where they would rather not have to redownload
-many megabytes repeatedly, e.g. hotel WiFi. :)  It stores both as `tgz` files
-in your `VAGRANT_HOME` directory, `~/.vagrant.d` by default. To enable this,
-either specify `VAGRANT_CACHE=1` on the command line or change the `CACHE`
-variable near the top of the Vagrantfile from `false` to `true`. **NOTE:** This
-will enable use of a Docker cache, but you must run `./docker_gen_cache.sh`
-from the `vagrant/` directory to initialize the cache. This is because
-creating the cache can take some time. It is recommended you pull any
-additional images you use to the master node, as this cache will be used by all
-nodes. For convenience, the GlusterFS and heketi images are automatically
-pulled to the master node.
+cache. This allows you to reuse those caches on subsequent provisioning of the
+VMs, and it is intended to help situations where one is developing in an
+environment where they would rather not have to redownload many megabytes
+repeatedly, e.g. hotel WiFi. :)  It stores both as `tgz` files in your
+`VAGRANT_HOME` directory, `~/.vagrant.d` by default. To enable this, either
+specify `VAGRANT_CACHE=1` on the command line or change the `CACHE` variable
+near the top of the Vagrantfile from `false` to `true`.
+
+**CUSTOM REGISTRY:** Similar to the caching feature, this environment supports
+interaction with a custom Docker registry. The idea is that a registry would
+be running in a local environment somewhere that could be used as a primary
+source for pulling container images into the VMs. Just specify the variable
+`custom_registry` in the `global_vars.yml` file to configure it. The following
+scripts are available to facilitate this feature:
+ * `docker-registry-run.sh`: A simple script to run a Docker registry in a
+   container on your local machine, listening on port 5000. **NOTE:** You may
+   need to open up the relevant firewall port on your local machine.
+ * `docker-cache.sh`: This script will detect all current images on a given VM
+   (default 'master') and push each image to the custom registry. Usage:
+   ```
+   ./docker-cache.sh 192.168.121.1:5000 master
+   ```
+A typical workflow to start using this would look like:
+ 1. Run `docker-registry-run.sh`
+ 2. Enable the custom registry in `global_vars.yml`.
+    * **NOTE:** Since the registry is currently empty, any search for container
+      images will proceed to the next registry (Docker Hub, by default).
+ 3. Start the vagrant environment
+ 4. Run `docker-cache.sh <host IP>:5000 master`
+ 5. Run `docker-cache.sh <host IP>:5000 node0`
+    * **OPTIONAL:** Run `docker pull gcr.io/google_containers/nginx-slim:0.8`
+      on `node0` before caching, since it is used in testing.
```

Comment:
why not run docker pull ....nginx before running any docker-cache.sh, and only afterwards run docker-cache.sh <IP>:5000 node0 (instead of using master) ?